src/app/directives/close-button.js

```diff
@@ -6,15 +6,15 @@
       restrict: 'E',
       templateUrl: 'directives/close-button.tpl.html',
       replace: true,
-      controller: ['$scope', '$rootScope', function($scope, $rootScope) {
+      scope: {
+        handleCloseClick: '&'
+      },
+      controller: ['$scope', function($scope) {
         $scope.close = function () {
-          var $inactiveElements = jQuery('.raml-console-tab').add('.raml-console-resource').add('li');
+          jQuery('.raml-console-tab.raml-console-is-active')
```

Comment:
I agree, I'll file issues for all of this problems so they can be fixed as part of another PR